Sources/Apodini/Semantic Model Builder/KnowledgeSources/DelegateFactory.swift

```diff
@@ -0,0 +1,29 @@
+//
+// This source file is part of the Apodini open source project
+//
+// SPDX-FileCopyrightText: 2019-2021 Paul Schmiedmayer and the Apodini project authors (see CONTRIBUTORS.md) <paul.schmiedmayer@tum.de>
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+
+/// A ``DelegateFactory`` allows for creating ``instance()``s of a ``Delegate``
+/// suitable for usage in an ``InterfaceExporter``.
+public struct DelegateFactory<H: Handler>: KnowledgeSource {
+    private let delegate: Delegate<H>
+    
+    public init<B>(_ blackboard: B) throws where B: Blackboard {
+        self.delegate = Delegate(blackboard[EndpointSource<H>].handler, .required)
+    }
+    
+    /// Creates one instance of the ``Delegate``.
+    ///
+    /// - Note: Use a new instance for each client-connection! Otherwise data may be shared between
+    /// all clients using the same instance.
+    public func instance() -> Delegate<H> {
```

Comment:
Maybe a bit picky (by a lot), but I think it might be a bit weird having the same method name `instance` use twice in similar context for a) in the DelegateFatory to return a `Delegate` instance and b) in the `Delegate` use to return the delegated `Handler` instance.
This creates the possibility of endpoint[DelegateFactory<H>.self].instance().instance().
Just something I stumbled upon when reading over the code where it got confused. Maybe its just me though 😁

Reply:
I kinda see your point, but then I actually don't. `endpoint[DelegateFactory.self].instance().instance()` should never occur (outside of tests) as there should be some other stuff going on in between instantiating a `Delegate` and preparing its wrapped child.
If you have a better suggestion for either of the two let me know.

Reply:
After all, I do not have a super strong opinion on that. And as you said, the use cases where each of those methods are used, are completely separate. Fair point. 👍